cmd/qspi/src/lib.rs

```diff
@@ -280,32 +283,49 @@ fn erase(
     core: &mut dyn Core,
     context: &mut HiffyContext,
     funcs: &HiffyFunctions,
-    sectors: &[u32],
+    all_sectors: &[u32],
 ) -> Result<()> {
     let f = funcs.get("QspiSectorErase", 1)?;
-    let mut ops = vec![];
 
     humility::msg!(
         "erasing {} bytes...",
-        sectors.len() as u32 * device.sector_size
+        all_sectors.len() as u32 * device.sector_size
     );
 
-    for addr in sectors {
-        if addr % device.sector_size != 0 {
-            bail!("illegal erase address 0x{:x}", addr);
-        }
+    //
+    // To unroll our loop, we need to know how many operations we can stuff in
+    // our text.  For this calculation, we'll be a bit pessimal -- we would
+    // actually expect to be able to serialize our per-sector operations in
+    // fewer than 16 bytes -- but we want to give ourselves a bit of slop.
```

Comment:
There's no real value in absorbing that complexity here, because the presence of the dummy requests actual eliminate the saving that one would get by having a more accurate (but aggressive) number. This estimate is designed to be conservative.

Reply:
I think I may have misunderstood the comment in that you may have been advocating composing the request but not actually sending it.  I still don't think it's worth the complexity delta because the payoff is pretty slight here, but that would definitely make more sense!